rest_framework_simplejwt/authentication.py

```diff
@@ -17,6 +18,19 @@
 )
 
 
+def enforce_csrf(request):
+    """
+    Enforce CSRF validation.
+    """
+    check = CSRFCheck()
+    # populates request.META['CSRF_COOKIE'], which is used in process_view()
+    check.process_request(request)
+    reason = check.process_view(request, None, (), {})
+    if reason:
+        # CSRF failed, bail with explicit error message
+        raise exceptions.PermissionDenied('CSRF Failed: %s' % reason)
```

Comment:
What if I want to pass csrf token only in header, not in header and cookie?

Reply:
This is probably a thing to ask silly ask, but isn't the idea behind csrf protection to defend users who have an active session? If we aren't using session authentication, how does csrf improve security?

Reply:
https://docs.djangoproject.com/en/3.0/ref/csrf/#ajax

In JS code we take csrf token from cookie as it is not rendered on template.